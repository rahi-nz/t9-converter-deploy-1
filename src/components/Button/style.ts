import styled from 'styled-components';
interface Prop {
  readonly disabled: boolean;
}
export const DialBtn = styled.button<Prop>`
  padding: 10px;
  border: 1px solid #bbbbbb;
  border-radius: 50%;
  width: 70px;
  height: 70px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  cursor: pointer;
  margin: 5px;
  background: #fff;
  ${props => {
    if (props.disabled)
      return `
          cursor: default;
        `;
  }}
`;

export const Title = styled.div`
  font-size: 18px;
`;

export const Subtitle = styled.div`
  font-size: 12px;
`;
