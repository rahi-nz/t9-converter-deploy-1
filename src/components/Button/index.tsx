import { DialBtn, Title, Subtitle } from './style';

export interface Props {
  handleClick: () => void;
  disabled: boolean;
  title: string;
  subtitle: string;
}

const Button = ({ handleClick, disabled, title, subtitle }: Props) => {
  return (
    <DialBtn onClick={handleClick} disabled={disabled}>
      <Title>{title}</Title>
      <Subtitle>{subtitle}</Subtitle>
    </DialBtn>
  );
};

export default Button;
