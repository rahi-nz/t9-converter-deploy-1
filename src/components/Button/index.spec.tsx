import renderer from 'react-test-renderer';
import Button from './index';

describe('<Button/>', () => {
  let rootElement;
  let testRenderer;
  const handleClick = jest.fn();
  // eslint-disable-next-line prefer-const,react/display-name
  rootElement = () => (
    <Button handleClick={handleClick} disabled={false} title="2" subtitle="abc" />
  );
  // eslint-disable-next-line prefer-const
  testRenderer = renderer.create(rootElement());

  it('Should render correctly', () => {
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
