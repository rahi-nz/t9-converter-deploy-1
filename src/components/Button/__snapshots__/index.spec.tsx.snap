// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> Should render correctly 1`] = `
<button
  className="style__DialBtn-niwgt9-0 dtWhVz"
  disabled={false}
  onClick={[MockFunction]}
>
  <div
    className="style__Title-niwgt9-1 iIrczW"
  >
    2
  </div>
  <div
    className="style__Subtitle-niwgt9-2 efnWUX"
  >
    abc
  </div>
</button>
`;
